drivers/sensor/lsm6dso/lsm6dso.c

```diff
@@ -268,6 +269,162 @@ static int lsm6dso_gyro_config(const struct device *dev,
 	return 0;
 }
 
+static int lsm6dso_activity_inactivity_config(const struct device *dev,
+			       enum sensor_channel chan,
+			       enum sensor_attribute attr,
+			       const struct sensor_value *val)
+{
+	const struct lsm6dso_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lsm6dso_tap_cfg2_t reg;
+	int32_t ret;
+
+	switch (attr) {
+	case SENSOR_ATTR_ACTIVITY_INACTIVITY_WAKE_THS:
+
+		if (lsm6dso_wkup_threshold_set(ctx, val->val1) < 0) {
+			LOG_ERR("Failed to set WKUP THS");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_ACTIVITY_INACTIVITY_WAKE_DUR:
+
+		if (lsm6dso_wkup_dur_set(ctx, val->val1) < 0) {
+			LOG_ERR("Failed to set WKUP DUR");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_ACTIVITY_INACTIVITY_SLEEP_DUR:
+
+		if (lsm6dso_act_sleep_dur_set(ctx, val->val1) < 0) {
+			LOG_ERR("Failed to set SLEEP DUR");
+			return -EIO;
+		}
+		break;
+
+	case SENSOR_ATTR_ACTIVITY_INACTIVITY_SLEEP_FUNC:
+
+		/* Enable and set the GYRO activity/inactivity (sleep) function. */
+		if (lsm6dso_act_mode_set(ctx, (lsm6dso_inact_en_t)val->val1) < 0) {
+			LOG_ERR("Failed to set INACT_EN bits");
+			return -EIO;
+		}
+
+		break;
+
+	case SENSOR_ATTR_ACTIVITY_INACTIVITY_ENABLE_BASIC_INT:
+
+		/* Enable basic interrupts (6D/4D, free-fall, wake-up, tap, inactivity). */
+		ret = lsm6dso_read_reg(ctx, LSM6DSO_TAP_CFG2, (uint8_t *)&reg, 1);
+		if (ret == 0) {
+			reg.interrupts_enable = val->val1;
+			ret = lsm6dso_write_reg(ctx, LSM6DSO_TAP_CFG2, (uint8_t*)&reg, 1);
+			if (ret < 0) {
+				LOG_ERR("Failed to set INTERRTUPT EN BIT");
+				return -EIO;
+			}
```

Comment:
Why didn't you use lsm6dso_pin_int2_route_set() here? Is there something that I'm missing?

Reply:
We are actually using only the last section of lsm6dso_pin_int2_route_set. Using lsm6dso_pin_int2_route_set was setting continuously the interrupt caused by a new sample coming from the internal temperature sensor.

Reply:
> lsm6dso_pin_int2_route_set

Mmmmh... Is that because you set 'val.drdy_temp' in 'lsm6dso_pin_int2_route_t' struct ?

Reply:
Can you explain to me where do I set that value in my patch?

Reply:
But you shouldn't get any interrupt from temp sensor. Are you sure the struct is inited to zero before setting the accel drdy?